src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -166,7 +166,28 @@ internal abstract class TexlFunction : IFunction
         public virtual bool ManipulatesCollections => false;
 
         // Return true if the function uses an input's column names to inform Intellisense's suggestions.
-        public virtual bool CanSuggestInputColumns => false;
+        public virtual bool CanSuggestInputColumns => false;
+
+        /// <summary>
+        /// If this returns false, the Intellisense will use Arg[0] type to suggest the type of the argument.
+        /// e.g. Collect(), Remove(), etc.
+        /// </summary>
+        /// <param name="argIndex"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public virtual bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)

```

Comment:
>TryGetTypeForArgSuggestionAt [](http://example.com/codeflow?start=28&length=28)

Does this apply to all functions? Or can this only be called in limited scenarios?

Reply:
Yes, for a consistent experience, we should suggest global on all functions. not just Collect(| etc...